lte/gateway/python/magma/mobilityd/rpc_servicer.py

```diff
@@ -233,12 +233,18 @@ def AllocateIPAddress(self, request, context):
                 composite_sid + ",ipv6", IPAddress.IPV6,
                 context, request,
             )
-            ipv4_addr = ipv4_response.ip_list[0]
-            ipv6_addr = ipv6_response.ip_list[0]
-            # Get vlan from IPv4 Allocate response
-            resp = AllocateIPAddressResponse(
-                ip_list=[ipv4_addr, ipv6_addr],
-                vlan=ipv4_response.vlan,
+            
+            try:
+                ipv4_addr = ipv4_response.ip_list[0]
```

Comment:
We should handle this error separately, given we use different allocators for IPv4 and IPv6, this means IPv4 can still be successful, with this current change we will throw away both of the allocations if one of them fails. 

How about checking for each list's content, if it's empty we will not access the list to avoid the `IndexError`, if it's not empty we will append the `ipv4_addr` or `ipv6_addr` (or both) content to the final `ip_list` and use that to set the content of `AllocateIPAddressResponse`. 

Wdyt?

Reply:
Sounds good to me! My initial motivation for this approach was that it might make it clearer to the caller that something went wrong. Also I wasn't sure if there is a scenario where someone requests both an IPv4 and an IPv6 address and can still reasonably proceed when only receiving one of the two.